backend/src/database/services/base-service.ts

```diff
@@ -0,0 +1,170 @@
+import type { UUID } from 'digital-fuesim-manv-shared';
+import type {
+    DataSource,
+    DeleteResult,
+    EntityManager,
+    EntityTarget,
+    FindManyOptions,
+    FindOneOptions,
+    FindOptionsWhere,
+} from 'typeorm';
+import { EntityNotFoundError } from 'typeorm';
+import { DatabaseError } from '../database-error';
+import type { BaseEntity } from '../entities/base-entity';
+import type { Creatable, Updatable } from '../dtos';
+
+export abstract class BaseService<Entity extends BaseEntity<Entity, any>> {
+    public constructor(protected readonly dataSource: DataSource) {}
+
+    /**
+     * @param initialObject `undefined` for creating a new object, {@link Entity} for updating an existing row.
+     */
+    protected abstract createSavableObject<TInitial extends Entity | undefined>(
+        initialObject: TInitial,
+        dto: TInitial extends Entity ? Updatable<Entity> : Creatable<Entity>,
+        manager: EntityManager
+    ): Entity | Promise<Entity>;
```

Comment:
Wouldn't this be better to split into two functions: `getCreateEntityObject` and `getUpdateEntityObject`?